pkg/chain/chain.go

```diff
@@ -65,9 +65,9 @@ type BondedECDSAKeepFactory interface {
 	GetKeepAtIndex(keepIndex *big.Int) (common.Address, error)
 }
 
-// BondedECDSAKeep is an interface that provides ability to interact with
-// BondedECDSAKeep ethereum contracts.
-type BondedECDSAKeep interface {
+// Keep is an interface that provides ability to interact with
+// keep ethereum contracts.
+type Keep interface {
```

Comment:
I see two possibilities:

1. We define separate interfaces for each of keep types and keep existing one to nest it in both.
```go
type FullyBackedBondedECDSAKeep interface {
  AbstractBondedECDSAKeep
}

type BondedECDSAKeep interface {
  AbstractBondedECDSAKeep
}

type ECDSAKeep interface {
// All the functions we already got in `BondedECDSAKeepFactory` interface
  OnSignatureRequested(
		keepAddress common.Address,
		handler func(event *SignatureRequestedEvent),
	) (subscription.EventSubscription, error)


	OnConflictingPublicKeySubmitted(
		keepAddress common.Address,
		handler func(event *ConflictingPublicKeySubmittedEvent),
	) (subscription.EventSubscription, error)

// etc....
}
```

2. We rename `BondedECDSAKeep` to `ECDSAKeep` (not just `Keep`).
```suggestion
type ECDSAKeep interface {
```